src/HttpKernel/ContaoCache.php

```diff
@@ -28,16 +27,18 @@ class ContaoCache extends HttpCache implements CacheInvalidation
     use EventDispatchingHttpCache;
 
     /**
-     * @param KernelInterface $kernel
-     * @param string|null     $cacheDir
+     * @param ContaoKernel $kernel
+     * @param string|null  $cacheDir
      */
-    public function __construct(KernelInterface $kernel, string $cacheDir = null)
+    public function __construct(ContaoKernel $kernel, string $cacheDir = null)
```

Comment:
I don't like this. Why don't we use `HttpCacheAware` instead?

Reply:
There is no `HttpCacheAware`?

Reply:
No? https://github.com/FriendsOfSymfony/FOSHttpCache/blob/master/src/SymfonyCache/HttpCacheAware.php

Reply:
That's a trait, you can't typehint a trait?

Reply:
Sure?

Reply:
Yes. You can't and even if you could, you shouldn't.

Reply:
But you also shall not type-hint against concrete classes, otherwise it is not possible to use another kernel class.

Reply:
Yeah, which is exactly what we want. You shouldn't be able to use a different kernel other than `ContaoKernel` for `ContaoCache`. We cannot typehint for an interface either :)

Reply:
> You shouldn't be able to use a different kernel other than `ContaoKernel`

Why not?

Reply:
Because it's our cache with our kernel? If you want to do something else, you're very likely not only adjusting the kernel but also the cache. Also, keep in mind that this is a managed edition.

Reply:
Both kernel and cache are set in the entry point and can therefore be adjusted even in the managed edition.

@contao/developers What's your opinion?